.github/workflows/reusable-build.yml

```diff
@@ -13,6 +13,8 @@ on:
 env:
   REGISTRY: ghcr.io
   FULL_REF: ${{ inputs.tag && format('refs/tags/{0}', inputs.tag) || github.ref }}
+  # This must match the entry in rustup-toolchain.toml at the repository root
+  RUSTUP_TOOLCHAIN: "1.81.0"
```

Comment:
I guess we could name this `RUST_TOOLCHAIN` to be consistent with the filename?

Also, could we read the appropriate value from the `rust-toolchain.toml` file here rather than having to keep these in sync?

Reply:
`RUSTUP_TOOLCHAIN` is the actual env var that's read by rust: https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file

And I guess we could extract this, but I didn't want to move away from dtolnay/rust-toolchain (at least for now), and parsing toml files in workflows can only be done through obscure github actions, which I didn't feel comfortable introducing.

Followup?